global/hooks/useEnvironmentalData/index.ts

```diff
@@ -173,32 +181,99 @@ const useEnvironmentalData = (origin: string) => {
 	 * @returns
 	 */
 	const formatUploadData = (submission: Submission): UploadData[] => {
-		const fileName = [submission.data.inserts?.sample.batchName];
 		const organization = submission.organization;
 		const submissionId = submission.id.toString();
-		return submission.data.inserts?.sample.records.reduce<UploadData[]>((acc, item, index) => {
-			// Retrieve the first error and indicate the number of additional errors if any
-			const errors = submission.errors.inserts?.sample || [];
+
+		// Formatting Insert records
+		const insertRecords = submission.data.inserts?.sample?.records?.map<UploadData>(
+			(item, index) => {
+				const fileName = [submission.data.inserts?.sample.batchName];
+				const errors = submission.errors.inserts?.sample || [];
+				const errorDetails = getErrorDetailsMessage(errors, index);
+				const identifier = item[NEXT_PUBLIC_ENVIRONMENTAL_SAMPLE_ID_FIELD_NAME]?.toString();
```

Comment:
env vars should be coming from the configs function used in other parts of this app, so they can be accounted for in the same central spot.